components/ws-manager-mk2/controllers/status.go

```diff
@@ -114,18 +117,11 @@ func (r *WorkspaceReconciler) updateWorkspaceStatus(ctx context.Context, workspa
 
 	switch {
 	case isPodBeingDeleted(pod):
-		workspace.Status.Phase = workspacev1.WorkspacePhaseStopping
-
-		if controllerutil.ContainsFinalizer(pod, workspacev1.GitpodFinalizerName) {
-			if isDisposalFinished(workspace) {
-				workspace.Status.Phase = workspacev1.WorkspacePhaseStopped
-			}
-
-		} else {
-			// We do this independently of the dispostal status because pods only get their finalizer
-			// once they're running. If they fail before they reach the running phase we'll never see
-			// a disposal status, hence would never stop the workspace.
+		if workspace.Status.Phase == workspacev1.WorkspacePhaseStopping && isDisposalFinished(workspace) {
 			workspace.Status.Phase = workspacev1.WorkspacePhaseStopped
```

Comment:
checking for the pod finalizer is old logic and no longer needed, as `isDisposalFinished` does a similar check to see if the workspace ever became ready and has anything to dispose of, so we can remove it here and simplify.